.github/workflows/deploy-all.yml

```diff
@@ -33,89 +34,180 @@ on:
+
       - name: Log in to OpenShift
         run: |
           oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
-      # Deploy SIMS-Api
+
       - name: Deploy SIMS-API
         working-directory: "./devops/"
         run: |
           make oc-deploy-api
-      # Deploy Workers
+
+  # Deploy workers.
+  deploy-workers:
+    name: Deploy Workers
+    runs-on: ubuntu-latest
+    needs: run-db-migrations
+    steps:
+      - name: Print env
+        run: |
+          echo BUILD NAMESPACE: $BUILD_NAMESPACE
+          echo BRANCH: ${{ github.event.inputs.gitRef }}
+          echo HOST_PREFIX: $HOST_PREFIX
+          
+      - name: Checkout Target Branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.gitRef }}
+
+      - name: Log in to OpenShift
+        run: |
+          oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
+
       - name: Deploy Workers
         working-directory: "./devops/"
         run: |
           make oc-deploy-workers
-      # Deploy Queues
+
+  # Deploy queue consumers.
+  deploy-queue-consumers:
+    name: Deploy Queue Consumers
+    runs-on: ubuntu-latest
+    needs: run-db-migrations
+    steps:
+      - name: Print env
+        run: |
+          echo BUILD NAMESPACE: $BUILD_NAMESPACE
+          echo BRANCH: ${{ github.event.inputs.gitRef }}
+          echo HOST_PREFIX: $HOST_PREFIX
+          
+      - name: Checkout Target Branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.gitRef }}
+
+      - name: Log in to OpenShift
+        run: |
+          oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
+
       - name: Deploy Queue Consumers
         working-directory: "./devops/"
         run: |
           make oc-deploy-queue-consumers
-      # Deploy Deploy Web/Frontend
+
+  # Deploy Web/Frontend.
+  deploy-web-frontend:
+    name: Deploy Web/Frontend
+    runs-on: ubuntu-latest
+    needs: run-db-migrations
+    steps:
+      - name: Print env
+        run: |
+          echo BUILD NAMESPACE: $BUILD_NAMESPACE
+          echo BRANCH: ${{ github.event.inputs.gitRef }}
+          echo HOST_PREFIX: $HOST_PREFIX
+          
+      - name: Checkout Target Branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.gitRef }}
+
+      - name: Log in to OpenShift
+        run: |
+          oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
+      
       - name: Deploy Web/Frontend
         working-directory: "./devops/"
         run: |
           make oc-deploy-web
+
+  # Deploy Camunda Definitions
   deployCamundaDefinitions:
     if: ${{ github.event.inputs.deployCamundaDefinitions == 'true' }}
     name: Deploy BPMNs and DMNs to Camunda
     needs:
-      - deploy
+      - deploy-sims-api
```

Comment:
Should needs consider workers also?

Reply:
May be queue consumers also.

Reply:
It is a good point. Maybe we can wait for all the deployments to be successful before deploying definitions.

Reply:
If I am not wrong,  they do not have any dependencies with any of the deployments. We can deploy it parallel if the inputs are checked. What you say?? @dheepak-aot @andrewsignori-aot @andrepestana-aot @sh16011993

Reply:
I believe that if we enforce that all the changes will be backward compatible (and we will as much as possible), we can have them deployed in parallel also.

Reply:
For Camunda definitions, assuming that we have a new worker task that will need an additional input parameter, the new worker app will need to be compatible with current and new versions of the bpmn, but the opposite may be not right.
My point is, maybe forms can be dependent on the API, and Camunda definitions may be dependent on workers.

Reply:
Also deploying Camunda definitions could be dependent on Queue consumers, where the structure of payload to start workflow instance or something like that may have changed and could be in sync with bpmn updates.

But we can discuss this in upcoming PR.